staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/validator_test.go

```diff
@@ -538,6 +542,221 @@ func TestValidate(t *testing.T) {
+					Reason:     forbiddenReason,
+				},
+			}, hasParamKind, nil),
+			attributes: newValidAttribute(nil, true),
+			params:     runtime.Object(nilUnstructured),
+			policyDecisions: []PolicyDecision{
+				generatedDecision(ActionDeny, "", metav1.StatusReasonForbidden),
+			},
+			authorizer: denyAll,
+		},
+		{
+			name: "test authorizer reason",
+			policy: getValidPolicy([]v1alpha1.Validation{
+				{
+					Expression: "authorizer.group('').resource('endpoints').check('create').reason() == 'fake reason'",
+					Reason:     forbiddenReason,
+				},
+			}, hasParamKind, nil),
+			attributes: newValidAttribute(nil, true),
+			params:     runtime.Object(nilUnstructured),
+			policyDecisions: []PolicyDecision{
+				generatedDecision(ActionAdmit, "", ""),
+			},
+			authorizer: denyAll,
+		},
+		{
+			name: "test authorizer allow path check",
+			policy: getValidPolicy([]v1alpha1.Validation{
+				{
+					Expression: "authorizer.path('/healthz').check('get').allowed()",
+					Reason:     forbiddenReason,
+				},
+				{
+					Expression: "authorizer.path('/healthz').check('get').denied() == false",
+					Reason:     forbiddenReason,
+				},
+			}, hasParamKind, nil),
+			attributes: newValidAttribute(nil, true),
+			params:     runtime.Object(nilUnstructured),
+			policyDecisions: []PolicyDecision{
+				generatedDecision(ActionAdmit, "", ""),
+				generatedDecision(ActionAdmit, "", ""),
+			},
+			authorizer: newAuthzAllowMatch(authorizer.AttributesRecord{
+				Path: "/healthz",
+				Verb: "get",
+			}),
+		},
+		{
+			name: "test authorizer not allowed path check",
+			policy: getValidPolicy([]v1alpha1.Validation{
+				{
+					Expression: "authorizer.path('/healthz').check('get').allowed()",
+					Reason:     forbiddenReason,
+				},
+			}, hasParamKind, nil),
+			attributes: newValidAttribute(nil, true),
+			params:     runtime.Object(nilUnstructured),
+			policyDecisions: []PolicyDecision{
+				generatedDecision(ActionDeny, "", metav1.StatusReasonForbidden),
+			},
+			authorizer: denyAll,
+		},
+		{
+			name: "test authorizer decision is denied path check",
+			policy: getValidPolicy([]v1alpha1.Validation{
+				{
+					Expression: "authorizer.path('/healthz').check('get').denied()",
+					Reason:     forbiddenReason,
+				},
+				{
+					Expression: "authorizer.path('/healthz').check('get').allowed() == false",
+					Reason:     forbiddenReason,
+				},
+			}, hasParamKind, nil),
+			attributes: newValidAttribute(nil, true),
+			params:     runtime.Object(nilUnstructured),
+			policyDecisions: []PolicyDecision{
+				generatedDecision(ActionAdmit, "", ""), // denied() return true, so the expression should evaluate to true
+				generatedDecision(ActionAdmit, "", ""),
+			},
+			authorizer: denyAll,
+		},
+		{
+			name: "test request resource authorizer allow check",
+			policy: getValidPolicy([]v1alpha1.Validation{
+				{
+					Expression: "authorizer.requestResource.check('reticulate').allowed()",
```

Comment:
I had to think about this one.  We should make an API with this verb just to confuse people.  :)

Reply:
I was using silly custom verb names while writing tests a 1:30am :smiley:   I'll swap in "custom-verb" so future readers can intuit what is meant here more easily :)